File: lab9Final_v2/lab9Gulyaev1.2/lab9Gulyaev1.2/ViewController.m
//
//  ViewController.m
//  Task1.2
//
//  Created by Pavel Bandiuk on 22.05.2020.
//  Copyright © 2020 Pavel Bandiuk. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIView *regV;
@property (weak, nonatomic) IBOutlet UITextField *loginL;
@property (weak, nonatomic) IBOutlet UITextField *emailL;
@property (weak, nonatomic) IBOutlet UITextField *passL;
@property (weak, nonatomic) IBOutlet UITextField *repassL;
@property (weak, nonatomic) IBOutlet UITextField *passR;
@property (weak, nonatomic) IBOutlet UITextField *loginR;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.regV setHidden:YES];
    NSUserDefaults* user = [NSUserDefaults standardUserDefaults];
    if([user objectForKey:@"login"] != nil || [user objectForKey:@"pass"] != nil) {
        [self performSegueWithIdentifier:@"closeZone" sender:self];
    }
}

- (IBAction)segment:(id)sender {
    UISegmentedControl *seg = (UISegmentedControl* ) sender;
    NSInteger select = seg.selectedSegmentIndex;
    
    if(select == 0) {			
        [self.regV setHidden:YES];
    } else {
        [self.regV setHidden:NO];
    }
}

- (IBAction)buttonL:(id)sender {
    if(![self.loginR.text isEqualToString:@""] || [self.passR.text isEqualToString:@""]) {
        NSUserDefaults* user = [NSUserDefaults standardUserDefaults];
        [user setObject:self.loginR.text forKey:@"login"];
        [user setObject:self.passR.text forKey:@"pass"];
        [self performSegueWithIdentifier:@"closeZone" sender:self];
    }
}

@end

File: lab9Final_v2/Task2.1/Task2.1/ViewController.swift
//
//  ViewController.swift
//  Task2.1
//
//  Created by Pavel Bandiuk on 22.05.2020.
//  Copyright © 2020 Pavel Bandiuk. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var switcher: UISwitch!
    @IBOutlet weak var segmentedControll: UISegmentedControl!
    @IBOutlet weak var regView: UIView!
    @IBOutlet weak var rePasswordL: UITextField!
    @IBOutlet weak var passwordL: UITextField!
    @IBOutlet weak var loginL: UITextField!
    @IBOutlet weak var passwordR: UITextField!
    @IBOutlet weak var loginR: UITextField!
    @IBOutlet weak var buttonOutlet: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.regView.isHidden = true
    }
    
    override func viewDidAppear(_ animated: Bool) {
        if ((UserDefaults.standard.string(forKey: "login") != nil) && (UserDefaults.standard.string(forKey: "password") != nil)) {
            print("Success")
            self.performSegue(withIdentifier: "mainVC", sender: self)
        }
    }
    @IBAction func segment(_ sender: Any) {
        if(segmentedControll.selectedSegmentIndex == 0) {
            regView.isHidden = true
             buttonOutlet.titleLabel?.text = NSLocalizedString("Sign in", comment: "")
        } else {
            regView.isHidden = false
            buttonOutlet.titleLabel?.text = NSLocalizedString("Sign up", comment: "")
        }
    }
    
    @IBAction func `switch`(_ sender: Any) {
    }
    
    @IBAction func buttonFunc(_ sender: Any) {
        if (switcher.isOn) {
            let login = loginL.text!
            let password = passwordL.text
            
            UserDefaults.standard.set(login, forKey: "login")
            UserDefaults.standard.set(password, forKey: "password")
            self.performSegue(withIdentifier: "mainVC", sender: self)
        } else {
            return
        }
    }
}

